components/econet/climate/econet_climate.cpp

```diff
@@ -48,142 +48,161 @@ climate::ClimateTraits EconetClimate::traits() {
 
 void EconetClimate::setup() {
   if (!current_temperature_id_.empty()) {
-    parent_->register_listener(current_temperature_id_, request_mod_, request_once_,
-                               [this](const EconetDatapoint &datapoint) {
-                                 current_temperature = fahrenheit_to_celsius(datapoint.value_float);
-                                 publish_state();
-                               });
+    parent_->register_listener(
+        current_temperature_id_, request_mod_, request_once_,
+        [this](const EconetDatapoint &datapoint) {
+          current_temperature = fahrenheit_to_celsius(datapoint.value_float);
+          publish_state();
+        },
+        false, this->src_adr_);
```

Comment:
If all calls now specify address, in a separate PR it would be good to remove its default value and change the order so you don't have to specify false for the other parameter.